### PR TITLE
Add pax and witnesses

### LINKED ISSUE #153: Can't list more than one injured Passenger
on incidents report

### PR DESCRIPTION
**Pull request by mboneil10**
Closes #153

### CODE REVIEW

**Comment by mboneil10** [comment]
Oh shit sorry I didn't notice I forgot to make witness tests

File: spec/system/drivers/editing_incidents_spec.rb [hunk 1]
```diff
@@ -7,18 +7,49 @@
...
       expect(page).to have_content 'Editing Driver Account of Incident'
       incident.destroy
       click_button 'Save report'
       wait_for_ajax!
-      expect(page).to have_selector 'p.notice', text: 'This incident report no longer exists.'
+      expect(page).to have_selector 'p.notice',
+                                    text: 'This incident report no longer exists.'
+    end
+  end
+  context 'adding multiple injured passengers' do
```
Comment (mboneil10) [inline spec/system/drivers/editing_incidents_spec.rb]: Actually, before I write any tests for witnesses. Should this test be in `spec/system/drivers/special_incident_fields_spec.rb`?

Comment (dfaulken) [reply spec/system/drivers/editing_incidents_spec.rb]: I don't remember why I made that file, but it's a bad filename (note the lack of a gerund helpfully describing the use case under test). What's in there?

Comment (mboneil10) [reply spec/system/drivers/editing_incidents_spec.rb]: It tests which fields are necessary/required

File: coverage/.last_run.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
   "result": {
-    "covered_percent": 88.48
+    "covered_percent": 34.94
```
Comment (Anbranin) [inline coverage/.last_run.json]: Didn't drop this much, did it?

File: spec/system/staff/viewing_incidents_spec.rb [hunk 3]
```diff
@@ -79,70 +79,86 @@
...
       visit incidents_url(mode: 'week')
       expect(page).to have_selector 'h2',
-        text: 'Sunday, December 31 — Saturday, January 6'
+                                    text: 'Sunday, December 31 — Saturday, January 6'
       click_button 'View for whole month'
       # Note that it doesn't go back to January,
       # because the first day of the week is in December.
       expect(page).to have_selector 'h2',
-        text: 'Friday, December 1 — Sunday, December 31'
+                                    text: 'Friday, December 1 — Sunday, December 31'
```
Comment (Anbranin) [inline spec/system/staff/viewing_incidents_spec.rb]: ew... why the massive indentation? Is this a rubocop thing?

Comment (mboneil10) [reply spec/system/staff/viewing_incidents_spec.rb]: Yes, should I remove it?

File: coverage/.last_run.json (hunk 2 above)
Comment (mboneil10) [reply coverage/.last_run.json]: I'm not sure. Something weird with my testing environment, I think.

Comment (mboneil10) [reply coverage/.last_run.json]: So I deleted my coverage directory. Then ran rspec. It didn't generate a new file. Idk.

File: spec/system/staff/viewing_incidents_spec.rb (hunk 3 above)
Comment (mboneil10) [reply spec/system/staff/viewing_incidents_spec.rb]: Ok

File: coverage/.last_run.json (hunk 2 above)
Comment (mboneil10) [reply coverage/.last_run.json]: Yes. Do you have a different suggestion? :(

File: spec/system/supervisors/editing_supervisor_reports_spec.rb [hunk 4]
```diff
@@ -29,4 +29,33 @@
...
+        fill_in 'Name', with: 'Karin'
+        fill_in 'Address', with: '51 Forestry Way, Amherst'
+      end
+      click_button 'Save supervisor report'
+
+      visit incident_url(incident)
+      expect(page).to have_selector 'h2', text: 'Supervisor Incident Report'
+      expect(page).to have_selector 'h3', text: 'Witness Information'
+      expect(page).to have_selector 'li',
+                                    text: 'Adam; 355 Governors Dr, Amherst'
```
Comment (dfaulken) [inline spec/system/supervisors/editing_supervisor_reports_spec.rb]: 255

File: app/assets/javascripts/incident_reports/edit.coffee [hunk 5]
```diff
@@ -17,7 +17,13 @@ determineShouldProvideReasonNotUpToCurb = ->
 
 addFields = (fieldsSelector) ->
   fields = $(fieldsSelector).last()
-  fields.clone().insertAfter fields
+  field = fields.clone()
+  for attrs in field.find('input')
```
Comment (dfaulken) [inline app/assets/javascripts/incident_reports/edit.coffee]: I think I'd prefer a method that takes some string containing a number and returns the same string with the number incremented.

Also, it's not a complete mystery where in the string the number will fall, right? Could we write a more specific (and thereby more descriptive) regex?

File: spec/system/supervisors/editing_supervisor_reports_spec.rb (hunk 4 above)
Comment (mboneil10) [reply spec/system/supervisors/editing_supervisor_reports_spec.rb]: Hah! Thanks

File: app/assets/javascripts/incident_reports/edit.coffee (hunk 5 above)
Comment (mboneil10) [reply app/assets/javascripts/incident_reports/edit.coffee]: Ok, I'll try

Comment (mboneil10) [reply app/assets/javascripts/incident_reports/edit.coffee]: @dfaulken why do you want the method to take that string and return the string with the number incremented? The reason I did this was to show that only the number was getting changed.

Comment (dfaulken) [reply app/assets/javascripts/incident_reports/edit.coffee]: Right. Agreed! I just think that the current code is a little cluttered with the combination of the `attrs` logic *and* the number-incrementing logic. Whereas if we abstract one into a function, we get clearer separation of concerns.

Comment (dfaulken) [reply app/assets/javascripts/incident_reports/edit.coffee]: Not a blocking issue - if you think that that's a step backwards in readability I'm happy to approve as is. It's only a few lines.

File: coverage/.last_run.json (hunk 2 above)
Comment (dfaulken) [reply coverage/.last_run.json]: At some point we should hold an intervention for @mboneil10 's `rspec` installation. This has been a recurring theme across a few projects, if I'm not mistaken.

File: app/assets/javascripts/incident_reports/edit.coffee (hunk 5 above)
Comment (mboneil10) [reply app/assets/javascripts/incident_reports/edit.coffee]: OHHH. A method. Got it.

File: spec/system/supervisors/editing_supervisor_reports_spec.rb [hunk 6]
```diff
@@ -29,4 +29,33 @@
...
+      click_on 'Edit Supervisor Report'
+      expect(page).to have_content 'Editing Supervisor Report'
+      check 'Were there witnesses?'
+      expect(page).to have_text 'Witness Information'
+      fill_in 'Name', with: 'Adam'
+      fill_in 'Address', with: '255 Governors Dr, Amherst'
+      click_button 'Add witness info'
+      second_field = all('.witness-fields')[1]
+      within second_field do
+        fill_in 'Name', with: 'Karin'
```
Comment (Anbranin) [inline spec/system/supervisors/editing_supervisor_reports_spec.rb]: IT'S-A-MEEEEE

**Pull request merged**